lib/wasix/src/syscalls/journal/mod.rs

```diff
@@ -31,9 +31,9 @@ pub struct JournalSyscallPlayer<'a, 'c> {
     pub ctx: FunctionEnvMut<'c, WasiEnv>,
     pub bootstrapping: bool,
 
-    pub journal_module_hash: Option<[u8; 8]>,
+    pub journal_module_hash: Option<Vec<u8>>,
```

Comment:
We can probably use a Boxed slice instead, as we'll need no conversion? (`Box<[u8]>`)

Reply:
My main concern was introducing an allocation here which both `Vec` and `Box` do unfortunately. `Vec` already derefs to a slice, so I don't think here is any advantage in using `Box<[u8]>`.

Reply:
Size will not be changed, so `Box<[u8]>` would probably be more optimal. I'm not religious about this, but as `Box<[u8]>` has a fixed slice, less memory will be allocated than for the vector.

This is a nit pick in any case

Reply:
Ah you're right. I didn't consider the size.